src/components/Explore/Explore.tsx

```diff
@@ -225,10 +220,16 @@ const Explore: React.FunctionComponent<{
     const hasData = some(chartSeries, ({ data }) => data.length > 0);
     const hasMultipleLocations = selectedLocations.length > 1;
 
-    const modalNormalizeCheckboxProps = {
-      hasMultipleLocations,
-      normalizeData,
-      setNormalizeData,
+    const [period, setPeriod] = useState<Period>(Period.ALL);
+    const dateRange = chartSeries.map(series =>
+      getDateRange(series.data, period),
+    );
```

Comment:
I think the period should be constant and derived from the current date, not dependent on the data

```ts
import { TimeUnit, subtractTime} from 'common/utils/time-utils';
// ...
const dateTo = new Date();
const dateFrom = subtractTime(dateTo, 60, TimeUnit.DAYS);
```